contest/views.py

```diff
@@ -17,65 +17,235 @@
+        'server_time':serverTime,'problem_pass_rate':problem_pass_rate},
+        context_instance = RequestContext(request, processors = [custom_proc]))
 
-    contest1 = {'name':'First contest','start_time':'2014/12/27 15:30:00','end_time':'2014/12/27 16:00:00',
-                    'contest_contestant':200,'contest_owner':'ma in joe','problem_list':problemList,'contestant_list':contestantList
-                    ,'clarification_list':clarificationList}
+def new(request):
+    if request.user.has_judge_auth():
+        if request.method == 'GET':
+            form = ContestForm()
+            return render(request,'contest/editContest.html',{'form':form})
+        if request.method == 'POST':
+            form = ContestForm(request.POST)
+            if form.is_valid():
+                new_contest = form.save()
+                logger.info('Contest: Create a new contest %s!' % new_contest.id)
+                return HttpResponseRedirect('/contest/')
+    else:
+        raise PermissionDenied
+    
 
-    return render(request, 'contest/contest.html',{'contest':contest1,'problem_list':problemList,
-        'contestant_list':contestantList,'server_time':serverTime},
-        context_instance = RequestContext(request, processors = [custom_proc]))
+def edit(request,contest_id):
+    contest = get_object_or_404(Contest,id = contest_id)
+    if user_info.has_c_ownership(request.user,contest):
+        if request.method == 'GET':        
+            contest_dic = model_to_dict(contest)
+            form = ContestForm(initial = contest_dic)
+            return render(request,'contest/editContest.html',{'form':form})
+        if request.method == 'POST':
+            form = ContestForm(request.POST, instance = contest)
+            if form.is_valid():
+                modified_contest = form.save()
+                logger.info('Contest: Modified contest %s!' % modified_contest.id)
+                return HttpResponseRedirect('/contest/')
+    else:
+        raise PermissionDenied
+
+def delete(request,contest_id):
+    try:
+        contest = Contest.objects.get(id = contest_id)
+    except Contest.DoesNotExist:
+        logger.warning('Contest: Can not delete contest %s! Contest not found!' % contest_id)
+        raise Http404("Contest does not exist, can not delete.")
+        
+    if user_info.has_c_ownership(request.user,contest):
+        deleted_contest_id = contest.id
+        contest.delete()
+        logger.info('Contest: Delete contest %s!' % deleted_contest_id)
+        return HttpResponseRedirect('/contest/')
+    else:
+        raise PermissionDenied
+
+def register(request,contest_id):
+    #check contest's existance
+    try:
+        contest = Contest.objects.get(id = contest_id)
+    except Contest.DoesNotExist:
+        logger.warning('Contest: Can not register contest %s! Contest not found!' % contest_id)
+        raise Http404("Contest does not exist, can not register.")
+    #check contestant existance
+    try:
```

Comment:
http://stackoverflow.com/questions/26617431/django-avoid-duplicates-in-many-to-many-intermediate-model
it seem like there is an function .exist()
but it may look alike.
FROM:
contestant = Contestant.objects.get(contest = contest,user = request.user)
TO:
Contestant.objects.get(contest = contest,user = request.user).exist()

if does exist, I would laterly use that contest object.

So in a word, I think it would be good to keep this(?